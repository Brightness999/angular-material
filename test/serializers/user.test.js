/**
 * @fileoverview Test logging of user data objects.
 */
const Logality = require('../..');
const { sink, stubLogality } = require('../lib/tester.lib');

const UDO_MOCK = {
  id: 10,
  email: 'one@go.com',
};

describe('User Data Object Logging', () => {
  stubLogality();

  test('Will log UDO Properly by default', done => {
    const logality = new Logality({
      appName: 'testLogality',
      wstream: sink(chunk => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world', { user: UDO_MOCK });
  });
});
