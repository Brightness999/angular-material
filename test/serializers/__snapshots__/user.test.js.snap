// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Data Object Logging Will log UDO Properly by default 1`] = `
Object {
  "context": Object {
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/serializers/user.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
    "user": Object {
      "email": "one@go.com",
      "id": 10,
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "severity": 6,
}
`;
